hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfspp/tools/hdfs-ownership.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LIBHDFSPP_TOOLS_HDFS_OWNERSHIP
+#define LIBHDFSPP_TOOLS_HDFS_OWNERSHIP
+
+#include <functional>
+#include <mutex>
+#include <optional>
+#include <utility>
+
+#include "hdfspp/status.h"
+
+namespace hdfs::tools {
+struct Ownership {
```

Comment:
This ownership concept is new.
What is the idea behind this?

Reply:
It's a struct to hold `user` and `group` names provided by the user as command line arguments. It's an internal data structure aiding in writing clean code.